espnet2/spk/encoder/rawnet3_encoder.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2023 Jee-weon Jung
+# Apache 2.0
+
+"""RawNet3 Encoder"""
+
+from typing import List, Optional, Tuple
+
+import torch
+import torch.nn as nn
+from asteroid_filterbanks import Encoder, ParamSincFB
+from typeguard import check_argument_types
+
+from espnet2.asr.encoder.abs_encoder import AbsEncoder
+from espnet2.spk.layers.RawNetBasicBlock import Bottle2neck, PreEmphasis
+
+
+class RawNet3Encoder(AbsEncoder):
+    """
+    RawNet3 encoder. Extracts frame-level RawNet embeddings from raw waveform.
+    paper: J. Jung et al., "Pushing the limits of raw waveform speaker
+        recognition", in Proc. INTERSPEECH, 2022.
+
+    Args:
+        block: type of encoder block class to use.
+        model_scale: scale value of the Res2Net architecture.
+        output_size: dimensionality of the hidden representation.
+        sinc_stride: stride size of the first sinc-conv layer where it decides
+            the compression rate (Hz).
+    """
+
+    def __init__(
+        self,
+        block=Bottle2neck,
```

Comment:
Better to use the string type to specify the `block`, because the vanilla yaml file does not support specifying Python modules as the value.

Reply:
I agree with you. But maybe we can leave this one because many preceding speaker of image models use this kind of definition and people might be used to it. 
But I can change as well :-)

Reply:
If it is configurable via yaml files, it will be fine. Can you make an example?

Reply:
@Emrys365, do you mean that we just make it a string like this?
```
  block: str = "bottle2neck",

  if block == "bottle2neck":
    self.block = Bottle2neck,
  else:
    raise ValueError(...)
```